pkg/datasources/external_functions_acceptance_test.go

```diff
@@ -1,80 +1,59 @@
 package datasources_test
 
 import (
-	"fmt"
+	"os"
 	"strings"
 	"testing"
 
+	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+
+	"github.com/hashicorp/terraform-plugin-testing/config"
 	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
 )
 
 func TestAcc_ExternalFunctions(t *testing.T) {
-	databaseName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
-	schemaName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
-	apiName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
-	externalFunctionName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
-	resource.ParallelTest(t, resource.TestCase{
-		Providers:    providers(),
+	if _, ok := os.LookupEnv("SKIP_EXTERNAL_FUNCTION_TESTS"); ok {
+		t.Skip("Skipping TestAcc_ExternalFunction")
+	}
+	accName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	m := func() map[string]config.Variable {
+		return map[string]config.Variable{
+			"database":                  config.StringVariable(acc.TestDatabaseName),
```

Comment:
if you are making these two optional, you should add 2 more tests

Reply:
i dont think 2 more tests are necessary, since there are only two possibilities: database + schema set, or neither set. I have added a test for the second case

Reply:
and what about just a database set?

Reply:
Oh, okay, we allow db only with schema, fine.